### DISCUSSION TITLE
After update to 2.1.2/2.1.3, Plausible is stuck in crash-loop with migrations error

### DISCUSSION THREAD

**Discussion by ItalyPaleAle** [plausible/analytics#4636] [Self-Hosted Support] [answered]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Using official Plausible Cloud hosting or self-hosting?

Self-hosting

### Describe the bug

After updating Plausible to the latest version (2.1.3), the application isn't starting up and is stuck in a crash loop:

```
34e8aa96895a Loading plausible..
34e8aa96895a Starting dependencies..
34e8aa96895a Starting repos..
34e8aa96895a Plausible.Repo database already exists
34e8aa96895a Plausible.IngestRepo database already exists
34e8aa96895a Creation of Db successful!
34e8aa96895a Loading plausible..
34e8aa96895a Starting dependencies..
34e8aa96895a Starting repos..
34e8aa96895a     truncate-location-data-table Done!
34e8aa96895a
34e8aa96895a ------------------------------------------------------------------------------
34e8aa96895a     create-location-data-table Done!
34e8aa96895a
34e8aa96895a ------------------------------------------------------------------------------
34e8aa96895a ** (MatchError) no match of right hand side value: {:error, %Ch.Error{code: 80, message: "Code: 80. DB::Exception: CREATE OR REPLACE TABLE query is supported only for Atomic databases. (INCORRECT_QUERY) (version 24.3.3.102 (official build))\n"}}
34e8aa96895a     (plausible 0.0.1) lib/plausible/data_migration/locations_sync.ex:117: Plausible.DataMigration.LocationsSync.run/0
34e8aa96895a     (ecto_sql 3.12.0) lib/ecto/migration/runner.ex:318: Ecto.Migration.Runner.perform_operation/3
34e8aa96895a     (stdlib 6.0) timer.erl:590: :timer.tc/2
34e8aa96895a     (ecto_sql 3.12.0) lib/ecto/migration/runner.ex:25: Ecto.Migration.Runner.run/8
34e8aa96895a     (ecto_sql 3.12.0) lib/ecto/migrator.ex:365: Ecto.Migrator.attempt/8
34e8aa96895a     (ecto_sql 3.12.0) lib/ecto/migrator.ex:282: anonymous fn/5 in Ecto.Migrator.do_up/5
34e8aa96895a     (ecto_sql 3.12.0) lib/ecto/migrator.ex:337: anonymous fn/6 in Ecto.Migrator.async_migrate_maybe_in_transaction/7
```

Not sure if it is relevant, but this setup has been running for a while (at least a few years).

This issue happened with 2.1.2 and 2.1.3. Rolling back to 2.1.1 worked.

### Expected behavior

Plausible should start up

### Environment

```markdown
- OS: Linux
- Platform: Podman
- Plausible: 2.1.2 and 2.1.3
- Clickhouse: `clickhouse/clickhouse-server:24.3.3.102-alpine`
```

**Comment by ruslandoga** [comment]
👋 @ItalyPaleAle 

Thank you for the report! Seems similar to (now fixed) https://github.com/plausible/analytics/discussions/4161 but in a different migration.

Failing migration: https://github.com/plausible/analytics/blob/v2.1.3/priv/ingest_repo/migrations/20240709181437_populate_location_data.exs

Failing step: https://github.com/plausible/analytics/blob/v2.1.3/lib/plausible/data_migration/locations_sync.ex#L117-L121

Failing query: https://github.com/plausible/analytics/blob/master/priv/data_migrations/LocationsSync/sql/update-location-data-dictionary.sql.eex

**Comment by ruslandoga** [comment]
According to https://clickhouse.com/docs/en/engines/database-engines/atomic "Atomic" engine is used by default, and the error says that you use a different database engine.

---

> Not sure if it is relevant, but this setup has been running for a while (at least a few years).

This seems relevant. Maybe you started using ClickHouse and created `plausible_events_db` before Atomic engine was the default one. That would be before version 20.10 which was released in Oct 2020.

**Comment by ruslandoga** [comment]
I'll move this to discussions since it's unlikely to be a problem with Plausible itself.

**Comment by ruslandoga** [comment, marked as answer]
https://github.com/ClickHouse/ClickHouse/pull/39933 is one way to resolve this issue.

**Reply by RealOrangeOne** [reply]
This seems to have fixed it for me - thanks!

**Reply by ItalyPaleAle** [reply]
This worked for me too, thanks!

One more link: https://kb.altinity.com/engines/altinity-kb-atomic-database-engine/how-to-convert-ordinary-to-atomic/
